pkg/cmd/adm/restart_test.go

```diff
@@ -1,216 +1,390 @@
-		newClient, fakeClient := NewFakeClients(t, deployment)
-		numberOfUpdateCalls := 0
-		fakeClient.MockUpdate = requireDeploymentBeingUpdated(t, fakeClient, namespacedName, 1, &numberOfUpdateCalls)
+	t.Run("rollout status for the deleted pods(operator) works", func(t *testing.T) {
+		//given
+		newClient, fakeClient := NewFakeClients(t, toolchainCluster, hostDeployment)
 		ctx := clicontext.NewCommandContext(term, newClient)
 
-		// when
-		err := restartHostOperator(ctx, fakeClient, cfg.OperatorNamespace)
+		//when
+		err := restartDeployment(ctx, fakeClient, "toolchain-host-operator", nil, testIOStreams,
+			func(ctx *clicontext.CommandContext, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, labelSelector string) error {
+				require.Equal(t, testIOStreams, ioStreams)
+				return nil
+			}, nil)
 
-		// then
+		//then
 		require.NoError(t, err)
-		AssertDeploymentHasReplicas(t, fakeClient, namespacedName, 1)
-		assert.Equal(t, 2, numberOfUpdateCalls)
 	})
 
-	t.Run("host deployment with the label is not present - restart fails", func(t *testing.T) {
-		// given
-		deployment := newDeployment(namespacedName, 1)
-		newClient, fakeClient := NewFakeClients(t, deployment)
-		numberOfUpdateCalls := 0
-		fakeClient.MockUpdate = requireDeploymentBeingUpdated(t, fakeClient, namespacedName, 1, &numberOfUpdateCalls)
+	t.Run("error in rollout status of the deleted pods(operator)", func(t *testing.T) {
+		//given
+		newClient, fakeClient := NewFakeClients(t, toolchainCluster, hostDeployment)
 		ctx := clicontext.NewCommandContext(term, newClient)
+		expectedErr := fmt.Errorf("Could not check the status of the deployment")
+		//when
+		err := restartDeployment(ctx, fakeClient, "toolchain-host-operator", nil, genericclioptions.NewTestIOStreamsDiscard(),
+			func(ctx *clicontext.CommandContext, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, labelSelector string) error {
+				return expectedErr
+			}, nil)
+
+		//then
+		require.EqualError(t, err, expectedErr.Error())
+	})
 
-		// when
-		err := restartHostOperator(ctx, fakeClient, cfg.OperatorNamespace)
+	t.Run("autoscalling deployment should not restart", func(t *testing.T) {
+		//given
+		newClient, fakeClient := NewFakeClients(t, toolchainCluster, memberDeployment, autoscalarDeployment)
+		ctx := clicontext.NewCommandContext(term, newClient)
+		//when
+		err := restartDeployment(ctx, fakeClient, "toolchain-member-operator", nil, genericclioptions.NewTestIOStreamsDiscard(),
+			func(ctx *clicontext.CommandContext, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, labelSelector string) error {
+				return nil
+			}, nil)
 
-		// then
-		require.Error(t, err)
-		AssertDeploymentHasReplicas(t, fakeClient, namespacedName, 1)
-		assert.Equal(t, 0, numberOfUpdateCalls)
+		//then
+		require.NoError(t, err)
+		require.Contains(t, term.Output(), "Found only autoscaling-buffer deployment in namespace toolchain-member-operator , which is not required to be restarted")
```

Comment:
but how do you know that it wasn't really restated? you don't intercept any calls there nor fail when it's called with the autoscaller-buffer deployment

Reply:
changed it